Bluetooth: conn: Expose simplified connection state in get_info

Thalley opened the pull request:
Add a state field in struct bt_conn_info that is a simplified
version of the internal state value (bt_conn_state_t).

This should provide an application to better determine the state
of the connection whne calling bt_conn_get_info, in case the
application does not keep track of the state itself.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

fixes https://github.com/zephyrproject-rtos/zephyr/issues/28009

Thalley committed:
Bluetooth: conn: Expose simplified connection state in get_info

Add a state field in struct bt_conn_info that is a simplified
version of the internal state value (bt_conn_state_t).

This should provide an application to better determine the state
of the connection whne calling bt_conn_get_info, in case the
application does not keep track of the state itself.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

asbjornsabo commented on include/bluetooth/conn.h:
```diff
@@ -309,6 +309,17 @@ enum {
+	/** Channel disconnected */
+	BT_CONN_STATE_DISCONNECTED,
+	/** Channel in connecting state */
+	BT_CONN_STATE_CONNECT,
```
I would prefer to add "ING" -> BT_CONN_STATE_CONNECTING
- That is what the documentation says
- It ensures that the name is not a subset of the next name (CONNECTED).

Ditto for DISCONNECT -> DISCONNECTING

Thalley replied:
I am not opposed to changing them either. If we decide to change them, so that change also be applied to the L2CAP, ISO and internal BT CONN states?

jhedberg replied:
Yes, it'd make sense to be consistent with the naming across the various layers and protocols

Thalley replied:
Will add additional commit to fix that

asbjornsabo replied:
Was that additional commit ever added?
I would like to have it.  (But if we decided not to do that, I will approve as is.)

Thalley replied:
It has not been added yet, will do that soon™

Thalley replied:
Added it now. Will do the L2CAP renaming in a different PR as that requires an API change.

Thalley replied:
https://github.com/zephyrproject-rtos/zephyr/pull/43165

Thalley committed:
Bluetooth: Host: Add `ing` to the bt_conn (dis)connect-ing state

Add `ing` to the `BT_CONN_CONNECT` and `BT_CONN_DISCONNECT`
states, so that the name better matches the actual state.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: ISO: Add `ing` to the bt_iso (dis)connect-ing state

Add `ing` to the `BT_ISO_CONNECT` and `BT_ISO_DISCONNECT`
states, so that the name better matches the actual state.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: ISO: Add `STATE` infix for ISO state enum

Change from BT_ISO_<state> to BT_ISO_STATE_<state>
to make the value more descriptive.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: ISO: Add named enum for bt_iso_state

Add an enum for the state field, to provide more
semantics to it.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

jhedberg commented on include/bluetooth/iso.h:
```diff
@@ -114,7 +114,7 @@ struct bt_iso_chan {
 	/** Channel QoS reference */
 	struct bt_iso_chan_qos		*qos;
-	uint8_t				state;
+	enum bt_iso_state		state;
```
Is it intentional that you’re increasing the struct member size here. We’ve used packed enums in some places where it matters.

Thalley replied:
IIRC the size of the enum should stay at 8-bit (given the possible values), right? 

In any case, this value is for the host only, and the size of it isn't used for any size-specific fields. 

I can __packed it, but I thought we generally wanted to avoid that in public APIs?

jhedberg replied:
AFAIK it’s always promoted to word size if you don’t use packed.

jhedberg replied:
Anyway, not an issue if it’s not critical to compact the struct here

Thalley replied:
> AFAIK it’s always promoted to word size if you don’t use packed.

From what I can tell, that's implementation (compiler) specific, and the only requirement is that an enum shall be able hold value _up to_ values representable by an int. 

In any case, this a minor case and I suggest to leave as is. I like the idea of using name enums as fields in the public API, as it gives a direct reference to values that may be used in the field

Thalley commented:
@jhedberg Will you please re-review?
